### PR TITLE
Fix issue 1295: failed visual-paste from registers >1

### PR DESCRIPTION
**Pull request by tomdl89**
The `kill-ring` let binding in `evil-visual-paste` meant that
`evil-get-register`'s view of `kill-ring` (used in `evil-paste-before`
and `evil-paste-after`) was limited to only entry `0` (and whatever
`evil-delete` may have added to rebound `kill-ring`). Therefore,
registers >1 were not actually visible to functions that needed
them. Condensing the `let*` binding so `kill-ring-yank-pointer` is
bound directly to `(list (current-kill 0))` was chosen as a simple and
effective method to fix the bug.

Fix for issue #1295

### CODE REVIEW

**Comment by tomdl89** [comment]
I'm not sure the CI ran properly:
`No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`

**Comment by TheBB** [comment]
Yeah, the way we're running CI has some problems. If an error occurs during test setup and not test running, then the backtrace won't be caught and Emacs will keep spinning forever.

In this case the problem is that you're missing a closing paren.

**Comment by tomdl89** [comment]
Ah I see - thanks for pointing that out! I didn't have smartparens on at the time. Now I have failing tests, hopefully I can fix.

**Comment by tomdl89** [comment]
Sorry it took me so long. All tests now pass (including one I added). @TheBB is there anything more you need from me? Thanks

**Comment by tomdl89** [comment]
It's there something I can do to help move this forward? Thanks

**Comment by TheBB** [comment]
I was about to merge a few days ago actually. I just had some problems understanding the test you wrote, and I couldn't reproduce it in vim.

Let me try again tomorrow.

**Comment by tomdl89** [comment]
Ah that's awesome, thanks. The test yanks 3 words then uses the `"3` register to visual paste over the first word of the next line. At least that's what I meant it to do! Let me know if you need me to give it another look.

**Comment by tomdl89** [comment]
I just tried replicating this in Neo Vim 0.5.0-427 and I couldn't. Looking at the docs, it seems that vim only populates the number registers >0 with what has been deleted, not yanked (http://vimdoc.sourceforge.net/htmldoc/change.html#quote_number). However, in emacs we `kill-ring-save` when we copy text, it populates the kill-ring when copied _and_ deleted. My change makes the behaviour more consistent with evil's (non-visual-) paste, although I can see that it is not strictly compatible with vim (although the current behaviour of visual paste just wiping the selected text on visual paste is neither compatible with vim nor emacs).

**Comment by TheBB** [comment]
Okay, that's fine. Could you put a comment to that effect in the test? Then I'll merge tomorrow.

**Comment by TheBB** [comment]
Thanks!
